.github/workflows/general_file_formatting_check.yml

```
@@ -0,0 +1,119 @@
+      - name: Check for trailing whitespace
+        id: whitespace_check
+        run: |
+          # Check for trailing whitespace in all text files
+          echo "🔍 Checking for trailing whitespace..."
+          WHITESPACE_FILES=$(grep -r --include="*.py" --include="*.yml" --include="*.yaml" --include="*.json" --include="*.md" --include="*.txt" --include="*.sh" --include="*.conf" -l '[[:space:]]$' . 2>/dev/null || true)
+
+          if [ -n "$WHITESPACE_FILES" ]; then
+            echo "whitespace_files<<EOF" >> $GITHUB_OUTPUT
+            echo "$WHITESPACE_FILES" >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+            
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)